### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by jstypka**
Some people complained in the issues sections about different problems, so I added small fixes and lumped them up in one pull request. Changes:

 - change the Travis settings so that it doesn't spam that much - only when builds fail
 - extracted hardcoded parameters from word2vec training to the `config.py`
 - added some nicer error checks in `main.py`
 - remove the CRNN model that didn't work anyway
 - substitute accuracy with top k accuracy which makes more sense for multi-label classification

They should be grouped into commits fairly reasonably, so it's easier to review.

@kaplun

### CODE REVIEW

File: magpie/main.py [hunk 1]
```diff
@@ -57,16 +57,22 @@ def train(self, train_dir, vocabulary, test_dir=None, callbacks=None,
         """
 
         if not self.word2vec_model:
-            print('word2vec model is not trained. Run train_word2vec() first.')
-            return
+            raise ValueError('word2vec model is not trained. ' + \
```
Comment (kaplun) [inline magpie/main.py]: All these I think should be `RuntimeError`s.

File: magpie/main.py [hunk 2]
```diff
@@ -57,16 +57,22 @@ def train(self, train_dir, vocabulary, test_dir=None, callbacks=None,
...
-            print('The scaler is not trained. Run fit_scaler() first.')
-            return
+            raise ValueError('The scaler is not trained. ' + \
+                             'Run fit_scaler() first.')
 
         if self.keras_model:
             print('WARNING! Overwriting already trained Keras model.')
 
+        if not os.path.isdir(train_dir):
+            print('The training directory ' + train_dir + ' does not exist')
```
Comment (kaplun) [inline magpie/main.py]: Is this a non failing error?

Comment (kaplun) [inline magpie/main.py]: Maybe you should print on `stderr`

File: magpie/nn/models.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def cnn(embedding_size, output_length):
     model.compile(
         loss='binary_crossentropy',
         optimizer='adam',
-        metrics=['accuracy'],
+        metrics=['top_k_categorical_accuracy'],
```
Comment (kaplun) [inline magpie/nn/models.py]: Can you add some more info in the commit log to explain the change? Is there some online reference you can point to?

File: magpie/nn/models.py [hunk 4]
```diff
@@ -12,8 +12,6 @@ def get_nn_model(nn_model, embedding, output_length):
         return cnn(embedding_size=embedding, output_length=output_length)
     elif nn_model == 'rnn':
         return rnn(embedding_size=embedding, output_length=output_length)
-    elif nn_model == 'crnn':
-        return crnn(embedding_size=embedding, output_length=output_length)
```
Comment (kaplun) [inline magpie/nn/models.py]: Can you add in the commit log why `crnn` is obsolete? Is there some online reference you can point to?

**Comment by jstypka** [comment]
valid comments @kaplun ! Will try to address and fix them when I have a minute!

**Comment by jstypka** [comment]
@kaplun I updated the code and added explanations in the commits. Check it out!
